adminpages/reports/login.php

```diff
@@ -252,194 +273,265 @@ function pmpro_report_login_page()
+
+		if(!isset($allvalues['thisdate']) || $thisdate != $allvalues['thisdate'])
+		{
+			$allvalues['today'] = 0;
+			$allvalues['thisdate'] = $thisdate;
+			$update = true;
+		}
 
-	update_option("pmpro_visits", $visits);
-}
-add_action("wp", "pmpro_report_login_wp_visits");
+		if(!isset($allvalues['thisweek']) || $thisweek != $allvalues['thisweek'])
+		{
+			$allvalues['week'] = 0;
+			$allvalues['thisweek'] = $thisweek;
+			$update = true;
+		}
 
-//we want to clear the pmpro_visit cookie on login/logout
-function pmpro_report_login_clear_visit_cookie()
-{
-	if(isset($_COOKIE['pmpro_visit']))
-		unset($_COOKIE['pmpro_visit']);
+		if(!isset($allvalues['thismonth']) || $thismonth != $allvalues['thismonth'])
+		{
+			$allvalues['month'] = 0;
+			$allvalues['thismonth'] = $thismonth;
+			$update = true;
+		}
+				
+		if(!isset($allvalues['thisyear']) || $thisyear != $allvalues['thisyear'])
+		{
+			$allvalues['ytd'] = 0;
+			$allvalues['thisyear'] = $thisyear;
+			$update = true;
+		}
+
+		if(!empty($update))
+			update_option('pmpro_' . $type, $allvalues);
+	}
+
+	return $allvalues;
 }
-add_action("wp_login", "pmpro_report_login_clear_visit_cookie");
-add_action("wp_logout", "pmpro_report_login_clear_visit_cookie");
 
-//track views
-function pmpro_report_login_wp_views()
+//track visits, views, and logins and save to user meta
+function pmpro_login_report_track_values($type, $user_id = NULL)
 {
 	//don't track admin
 	if(is_admin())
-		return;
+		return false;
+
+	//default to current user
+	if(empty($user_id))
+	{
+		global $current_user;
+		$user_id = $current_user->ID;
+	}
+
+	//need a type
+	if(empty($type))
+		return false;
 
-	global $current_user;
+	//check for cookie for visits
+	if($type == "visits" && !empty($_COOKIE['pmpro_visit']))
+		return false;
+
+	//set cookie for visits
+	if($type == "visits" && empty($_COOKIE['pmpro_visit']))
+		setcookie("pmpro_visit", "1", NULL, COOKIEPATH, COOKIE_DOMAIN, false);	
+
+	//some vars for below
 	$now = current_time('timestamp');
+	$thisdate = date("Y-d-m", $now);
+	$thisweek = date("W", $now);
+	$thismonth = date("n", $now);
+	$thisyear = date("Y", $now);
 
-	//track for user
-	if(!empty($current_user->ID))
+	//track user stats if we have one
+	if(!empty($user_id))
 	{
-		$views = $current_user->pmpro_views;
-		if(empty($views))
-			$views = array("last"=>"N/A", "month"=>0, "alltime"=>0);
-
-		//track logins for user
-		$views['last'] = date_i18n(get_option("date_format"), $now);
-		$views['alltime'] = $views['alltime'] + 1;
-		$thismonth = date_i18n("n", $now);
-		if(isset($views['thismonth']) && $thismonth == $views['thismonth'])
-			$views['month'] = $views['month'] + 1;
-		else
+		//get values
+		$values = pmpro_login_report_get_values_for_user($type, $user_id);
+
+		if($values !== false)
 		{
-			$views['month'] = 1;
-			$views['thismonth'] = $thismonth;
+			//track for user
+			$values['last'] = date(get_option("date_format"), $now);
+			$values['alltime']++;

```

Comment:
Change all of these ++ to something like:

`$values['alltime'] = $values['alltime'] + 1;`

There is a note RE a PHP bug fix on these in the old version.